### PR TITLE
Dev task/controllers unit tests

### PR DESCRIPTION
**Pull request by bendiggle**
Unit tests for the controllers that are used in gro have been added

### CODE REVIEW

**Comment by JoeChapman** [comment]
@bendiggle can you rewrite these tests with ES6 please, arrow functions, `const`, and `let`.

**Comment by JoeChapman** [comment]
@bendiggle good work though

**Comment by bendiggle** [comment]
@JoeChapman sure, will do that now.

**Comment by bendiggle** [comment]
@JoeChapman is this ok? I'm still quite new to ES6

File: test/unit/apps/gro/controllers/confirm.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  'path': path,
+  '../../common/models/email': Model
+});
+
+describe('apps/gro/controllers/confirm', () => {
+
+  describe('.saveValues()', () => {
+
+    let controller;
+    let emailText = 'email@email.com';
```
Comment (JoeChapman) [inline test/unit/apps/gro/controllers/confirm.js]: `const`

**Comment by bendiggle** [comment]
@JoeChapman That's now been changed to const

File: test/unit/apps/gro/controllers/confirm.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+    let req = {
+      sessionModel: {
```
Comment (JoeChapman) [inline test/unit/apps/gro/controllers/confirm.js]: `const`

File: test/unit/apps/gro/controllers/confirm.js [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+        attributes: {
+          'email-text': 'email@email.com'
+        }
+      },
+    };
+    let res = {};
+    let callback = sinon.stub();
+
```
Comment (JoeChapman) [inline test/unit/apps/gro/controllers/confirm.js]: `const`
`const`

File: test/unit/apps/gro/controllers/confirm.js [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+      BaseController.prototype.saveValues = sinon.stub();
+      controller.saveValues(req, res, callback);
+    });
+
+    it('calls the email Model', () => {
+      Model.prototype.save.should.have.been.called;
+    });
+
+    it('saves the email to the session', function () {
+      let sessionData = Model.args[0][0];
```
Comment (JoeChapman) [inline test/unit/apps/gro/controllers/confirm.js]: `const`

File: test/unit/apps/gro/controllers/confirm.js [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+      sessionData.should.have.property('email').and.be.equal(emailText);
+    });
+    //
```
Comment (JoeChapman) [inline test/unit/apps/gro/controllers/confirm.js]: remove comments

File: test/unit/apps/gro/controllers/post.js [hunk 6]
```diff
@@ -0,0 +1,50 @@
...
+    controllers: {
+      base: BaseController
+    }
+  }
+});
+
+describe('apps/gro/controllers/post', () => {
+
+  let controller;
+  let req = {
```
Comment (JoeChapman) [inline test/unit/apps/gro/controllers/post.js]: `const`

File: test/unit/apps/gro/controllers/when.js [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+    }
+  }
+});
+
+describe('apps/gro/controllers/when', () => {
+
+  describe('instantiated', () => {
+
+    let controller;
+    let args = {template: 'index'};
```
Comment (JoeChapman) [inline test/unit/apps/gro/controllers/when.js]: `const`

**Comment by JoeChapman** [comment]
@bendiggle nice work, cool

**Comment by bendiggle** [comment]
@JoeChapman Changed the rest of them now too

**Comment by JoeChapman** [comment]
Tests are failing @bendiggle

**Comment by bendiggle** [comment]
I@JoeChapman Yep, yep. On this occasion I didn't, so when I deleted the comments (//) it created some white-spacing lint error.
They have passed now!

**Pull request merged**